### PR TITLE
add second player

### PR DESCRIPTION
**Pull request by kudukacar**

### CODE REVIEW

File: lib/tic_tac_toe/human_player.ex [hunk 1]
```diff
@@ -7,30 +7,30 @@ defmodule TicTacToe.HumanPlayer do
...
-        Display.output(%{state: state} = display, error)
+        Display.output(display, error)
 
-        selection(%HumanPlayer{
-          token: token,
-          display: %{display | state: List.delete_at(state, 0)},
-          parser: parser,
-          validator: validator
-        })
+        selection(%{player | display: display}, board)
```
Comment (damonkelley) [inline lib/tic_tac_toe/human_player.ex:33]: 🎉

File: test/tic_tac_toe_test/human_player_test.exs [hunk 2]
```diff
@@ -11,41 +11,51 @@ defmodule TicTacToe.HumanPlayerTest do
...
+    defstruct [:input, :moves]
 
     defimpl Display do
       def output(_console, _message) do
       end
 
-      def input(%DisplayWithInputs{state: state}, _message) do
-        List.first(state)
+      def input(%DisplayWithInputs{moves: moves} = display, _message) do
+        %{display | input: List.first(moves), moves: List.delete_at(moves, 0)}
```
Comment (damonkelley) [inline test/tic_tac_toe_test/human_player_test.exs]: Hey! This is really common pattern that you can achieve with pattern matching

```suggestion
      def input(%DisplayWithInputs{moves: [input | moves]} = display, _message) do
        %{display | input: input, moves: moves}
```

File: lib/tic_tac_toe.ex [hunk 3]
```diff
@@ -1,28 +1,31 @@
...
-    new_display = %{display | state: show_board(board, presenter, display)}
-    play_turn(player, board) |> show_board(presenter, new_display)
+  def run(state) do
+    state
+    |> show_board()
+    |> play_turn()
+    |> show_board()
+    |> play_turn()
+    |> show_board()
+    |> Map.get(:display)
```
Comment (damonkelley) [inline lib/tic_tac_toe.ex]: Do we need to return the display? Or can we return the `state`?

File: lib/mix/tasks/tic_tac_toe.ex [hunk 4]
```diff
@@ -1,7 +1,15 @@
...
+  alias TicTacToe.HumanPlayer
 
   def run(_) do
-    TicTacToe.run()
+    TicTacToe.run(%{
+      board: List.duplicate(nil, 9),
+      display: %Console{},
+      presenter: Presenter,
+      players: [%HumanPlayer{token: "X"}, %HumanPlayer{token: "O"}]
+    })
```
Comment (damonkelley) [inline lib/mix/tasks/tic_tac_toe.ex:13]: Nice!!!

File: lib/tic_tac_toe/board.ex [hunk 5]
```diff
@@ -11,4 +12,9 @@ defimpl TicTacToe.Board, for: List do
   def place_token(board, position, token) do
     List.replace_at(board, position - 1, token)
   end
+
+  def available_positions(board) do
+    Enum.to_list(1..length(board))
+    |> Enum.filter(fn x -> get(board, x) == nil end)
```
Comment (damonkelley) [inline lib/tic_tac_toe/board.ex:18]: Cool

File: test/tic_tac_toe_test/human_player_test.exs (hunk 2 above)
Comment (kudukacar) [reply test/tic_tac_toe_test/human_player_test.exs]: Nice!  I forgot you showed me this trick.

File: lib/tic_tac_toe.ex (hunk 3 above)
Comment (kudukacar) [reply lib/tic_tac_toe.ex]: We can return state.  I'll adjust.

**Pull request merged**